detox/src/artifacts/screenshot/SimulatorScreenshotPlugin.js

```diff
@@ -38,23 +38,26 @@ class SimulatorScreenshotPlugin extends ScreenshotArtifactPlugin {
   }
 
   async onBeforeUninstallApp(event) {
-    const snapshots = [
-      ...Object.values(this.snapshots.fromTest),
-      ...Object.values(this.snapshots.fromSession)
-    ];
+    await this.api.requestIdleCallback(async () => {
```

Comment:
The problem here is that all the screenshot and video artifacts are being saved in an unhurried one-by-one and very asynchronous way (historically, due to Android's `adb pull` considerations) → that would mean that at the time we get into this callback, there would be some delayed move/copy operations which cannot be easily accessed.

This is why I am forcing the plugin to block the main flow execution, so that all the pending operations are completed and then I continue blocking the flow until all the remaining artifacts get relocated to a safe location (outside of the application folder).

Only after that we resume the flow.